File: app/src/main/java/pl/mosenko/sunnypodlaskie/broadcastreceiver/DeviceBroadcastReceiver.java
package pl.mosenko.sunnypodlaskie.broadcastreceiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import pl.mosenko.sunnypodlaskie.util.WeatherDataAlarmSyncUtil;

/**
 * Created by syk on 26.05.17.
 */

public class DeviceBroadcastReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (Intent.ACTION_BOOT_COMPLETED.equals(action)) {
            WeatherDataAlarmSyncUtil.startAlarm(context);
        }
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/broadcastreceiver/DeviceBroadcastReceiver.kt
package pl.mosenko.sunnypodlaskie.broadcastreceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject
import pl.mosenko.sunnypodlaskie.util.WeatherAlarmSyncUtil

/**
 * Created by syk on 26.05.17.
 */
class DeviceBroadcastReceiver : BroadcastReceiver(), KoinComponent {

    private val weatherAlarmSyncUtil: WeatherAlarmSyncUtil by inject()

    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            weatherAlarmSyncUtil.startAlarm()
        }
    }
}
